builtInFunctions/esdtDataStorage.go

```diff
@@ -383,7 +383,16 @@ func (e *esdtDataStorage) saveESDTMetaDataToSystemAccount(
 	}
 
 	currentSaveData, _, err := systemAcc.AccountDataHandler().RetrieveValue(esdtNFTTokenKey)
-	if !mustUpdate && len(currentSaveData) > 0 {
+	if err != nil {
+		return err
+	}
+
+	err = e.saveMetadataIfRequired(esdtNFTTokenKey, systemAcc, currentSaveData, esdtData)
+	if err != nil {
```

Comment:
Interesting that in the old version the error handling was not done, and it seems that it was based only on checking the length of the currentSaveData variable returned. This change could create now backwards compatibility?

Reply:
no, searched through the data trie tracker's get method: it returned errors only if:
1. the data managed in the internal map got corrupted, somehow (this is a game-over turning point for the node, anyway)
2. the data retrieved from the storage can not be fetched (because of an error, not because it was missing)

Reply:
ok

Reply:
reverted